### PR TITLE
db: add flush and properties

### PR DESCRIPTION
**Pull request by BusyJay**

### CODE REVIEW

File: tirocks/src/db/properties.rs [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+    util::{self, ffi_call, range_to_rocks},
+    RawDb, Result, Status,
+};
+
+use super::cf::RawCfHandle;
+
+impl RawDb {
+    /// DB implementations can export properties about their state via this method. If "prop"
+    /// is a valid property understood by this DB implementation (see struct inherits `Property`
+    /// trait for valid options), returns true. Otherwise, returns false.
```
Comment (tabokie) [inline tirocks/src/db/properties.rs]: Wrong doc for return value.

File: tirocks/src/db/flush.rs [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+                input_file_names.as_ptr(),
+                input_file_names.len(),
+                output_level,
+            ))
+        }
+    }
+
+    /// This function will wait until all currently running background processes
+    /// finish. After it returns, no background process will be run until
+    /// [`continue_background_work`] is called
```
Comment (tabokie) [inline tirocks/src/db/flush.rs]:
```suggestion
    /// [`continue_background_work`] is called.
```

**Review by tabokie — APPROVED** [review]

File: tirocks/src/db/flush.rs [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+
+    /// Flushes multiple column families.
+    ///
+    /// If atomic flush is not enabled, `flush_multi` is equivalent to calling [`flush`] multiple
+    /// times. If atomic flush is enabled, `flush_multi` will flush all column families specified
+    /// in 'cfs' up to the latest sequence number at the time when flush is requested.
+    /// Note that RocksDB 5.15 and earlier may not be able to open later versions with atomic
+    /// flush enabled.
+    #[inline]
+    pub fn flush_multi(&self, option: &FlushOptions, cfs: &[&RawCfHandle]) -> Result<()> {
```
Comment (Connor1996) [inline tirocks/src/db/flush.rs]: How about `flush_cfs`?

File: tirocks/src/db/flush.rs [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+                files.as_ptr(),
+                files.len(),
+                opt.as_ptr(),
+            ))
+        }
+    }
+
+    /// Same as [`ingest_external_file`] but avoid blocking writes with best effort.
+    #[inline]
+    pub fn ingest_external_file_optimized(
```
Comment (Connor1996) [inline tirocks/src/db/flush.rs:185]: How about using it as the impl of `ingest_external_file`

Comment (BusyJay) [reply tirocks/src/db/flush.rs:185]: It might not be a good idea as it would introduce a surprise for people who were familiar with rocksdb.

Comment (Connor1996) [reply tirocks/src/db/flush.rs:185]: but it's just an optimization without breaking any behavior

Comment (BusyJay) [reply tirocks/src/db/flush.rs:185]: It will flush if overlap.

**Review by Connor1996 — APPROVED** [review]

**Pull request merged**